django_mongodb/client.py

```diff
@@ -4,11 +4,51 @@
 
 
 class DatabaseClient(BaseDatabaseClient):
-    executable_name = "mongo"
+    executable_name = "mongosh"
+    help_options = ("--help", "-h")
 
     @classmethod
     def settings_to_cmd_args_env(cls, settings_dict, parameters):
-        raise NotImplementedError
+        options = settings_dict.get("OPTIONS", {})
+        uri = options.get("uri")
+        args = [cls.executable_name]
+        if not uri:
+            host = settings_dict["HOST"]
+            port = settings_dict["PORT"]
+            if port and host:
+                host += f":{port}"
+
+            dbname = settings_dict["NAME"]
+            user = settings_dict["USER"]
+            passwd = settings_dict["PASSWORD"]
+            auth_database = options.get("authenticationDatabase")
```

Comment:
The pymongo kwarg seems to be `authSource`.
```
>>> MongoClient(authenticationDatabase='foo')
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/home/tim/.virtualenvs/django312/lib/python3.12/site-packages/pymongo/mongo_client.py", line 806, in __init__
    dict(common.validate(keyword_opts.cased_key(k), v) for k, v in keyword_opts.items())
  File "/home/tim/.virtualenvs/django312/lib/python3.12/site-packages/pymongo/mongo_client.py", line 806, in <genexpr>
    dict(common.validate(keyword_opts.cased_key(k), v) for k, v in keyword_opts.items())
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tim/.virtualenvs/django312/lib/python3.12/site-packages/pymongo/common.py", line 821, in validate
    value = validator(option, value)
            ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tim/.virtualenvs/django312/lib/python3.12/site-packages/pymongo/common.py", line 167, in raise_config_error
    raise ConfigurationError(f"Unknown option {key}")
pymongo.errors.ConfigurationError: Unknown option authenticationDatabase
```

Reply:
Well, mongosh runs with other parameters:
--authenticationDatabase instead of authSource. am I using an old version of mongosh?
Shall I change it in order to be more consistent?
I am using mongosh version 2.1.1

Reply:
I think you missed the point [I tried to make before](https://github.com/timgraham/django-mongodb/pull/1#discussion_r1589490189). Whatever is in `OPTIONS` is passed directly to `MongoClient`.
https://github.com/mongodb-labs/django-mongodb/blob/28f52b14bc6eab9eca3bebaa7241e46848deab90/django_mongodb/base.py#L85-L95
So you need to pass `--authenticationDatabase` to `mongosh`, but fetch `authSource` from `OPTIONS`.

Reply:
Ok, I got it. django_mongo/base and the dbshell read from the same settings (which is very reasonable). Ok, I will adapt my script to those parameters.

Reply:
Done!